packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -11,14 +11,14 @@ const path = require('path');
 const crypto = require('crypto');
 const _ = require('lodash');
 const util = require('util');
-const filenamify = require('filenamify');
+const slugify = require('@sindresorhus/slugify');
 const mime = require('mime-types');
 
 const { bytesToKbytes } = require('../utils/file');
 
 const randomSuffix = () => crypto.randomBytes(5).toString('hex');
 const generateFileName = name => {
-  const baseName = filenamify(name, { replacement: '_' }).replace(/\s/g, '_');
+  const baseName = slugify(name, { separator: '_', lowercase: false });
```

Comment:
I'm sorry I wasn't clear enough :/ 

Can you update the naameToSlug function to accept a second parameter so you don't need to add the lib as a dependency but use strapi-utils :)